build/lpc55pins/src/lib.rs

```diff
@@ -98,12 +127,13 @@ impl ToTokens for PinConfig {
     fn to_tokens(&self, tokens: &mut TokenStream) {
         let final_pin = self.pin.to_token_stream();
         let alt_num = format_ident!("Alt{}", self.get_alt());
-        let mode = format_ident!("{}", self.get_mode());
-        let slew = format_ident!("{}", self.get_slew());
-        let invert = format_ident!("{}", self.get_invert());
-        let digimode = format_ident!("{}", self.get_digimode());
-        let od = format_ident!("{}", self.get_opendrain());
 
+        let mode = format_ident!("{}", format!("{:?}", self.get_mode()));
+        let slew = format_ident!("{}", format!("{:?}", self.get_slew()));
+        let invert = format_ident!("{}", format!("{:?}", self.get_invert()));
+        let digimode =
+            format_ident!("{}", format!("{:?}", self.get_digimode()));
+        let od = format_ident!("{}", format!("{:?}", self.get_opendrain()));
```

Comment:
can't these just be
```suggestion
        let mode = format_ident!("{:?}", self.get_mode());
        let slew = format_ident!("{:?}", self.get_slew());
        let invert = format_ident!("{:?}", format!("{:?}", self.get_invert());
        let digimode =
            format_ident!("{:?}", self.get_digimode());
        let od = format_ident!("{:?}", self.get_opendrain());
```
AFAICT, we shouldn't need the second string allocation?

Reply:
also, it occurs to me an alternative to using the `Debug` impl could be implementing `ToTokens` for the various enums, but...that would probably just internally use `syn::Ident::new` or something, which would end up looking similarish. But, you could hard-code the string, which is probably either better or worse...

Reply:
Oh, huh, I was apparently wrong about how `format_ident!` works!

I suppose another option would just be to write
```rust
syn::Ident::new(format!("{:?}", self.get_mode()), proc_macro2::Span::call_site())
```
to avoid the second allocation, although this is, admittedly, uglier...

Reply:
Yeah, I'm on the fence here; if all solutions are equally ugly, then might as well pick the shorter one.

Reply:
Yeah...the impact on build time is probably not meaningful. Carry on.